mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -152,9 +152,10 @@ struct LinearizeVectorExtractStridedSlice final
   matchAndRewrite(vector::ExtractStridedSliceOp extractOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
     Type dstType = getTypeConverter()->convertType(extractOp.getType());
```

Comment:
Why don't we just cast dstType to ShapedType here? I think it carries more information/methods v.s. `Type`. And you don't need to cast it in the below if condition.

Reply:
Could you please provide more detail on how ShapedType would not need a cast for `cast<VectorType>(dstType).isScalable()`? AFAIK ShapedType has no `isScalable()`, I do not see other places in the pattern where we could use ShapedType's information/methods that are not provided by Type.

Reply:
Sorry for the confusion, what I meant is `VectorType`. So it could either be 

```
VectorType dstType = getTypeConverter()->convertType(extractOp.getType());
```

or

```
auto dstType = cast<VectorType>(getTypeConverter()->convertType(extractOp.getType()));
```

Reply:
you can actually do `getTypeConverter()->convertType<VectorType>(...)`. Also, it's better to check `convertType` result for null.

Reply:
Thanks for the clarification, should now be addressed